controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,144 @@
+ */
+ 
+// Special thanks to Hamburg Bit-Bots for ideas for this controller
+// https://github.com/bit-bots/wolfgang_robot/blob/feature/recognition/wolfgang_webots_sim/src/wolfgang_webots_sim/webots_camera_controller.py
+
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <math.h>
+#include <random>
+#include <webots/Supervisor.hpp>
+
+#include "yaml-cpp/yaml.h"
+
+// AxisAngle rotations will be read from a config file and saved here
+std::vector<std::array<double, 4>> rotations;
+double minDistance;
+double zHeight;
+double xSize;
+double ySize;
+
+int main(int argc, char** argv) {
+
+    // Make sure we have the command line arguments we need. At a minimum we should have the def argument
+    // of the robot supervisor. Other arguments are def arguments of robots that should be controlled by
+    // this supervisor
+    if (argc < 2) {
+        std::cerr << "Usage: " << argv[0] << " <DEF>" << std::endl;
+        return EXIT_FAILURE;
+    }
+    // Load self-def argument which will be used to identify the robot using the webots getFromDef function
+    std::string def = argv[1];
+
+    // Load def arguments of other robots in the field into a vector
+    std::vector<webots::Node*> otherRobotsNodes;
+    // Only one supervisor instance can exist in a single controller. As such the same one will be used
+    // every time
+    webots::Supervisor supervisor = webots::Supervisor();
+    for (int i = 2; i < argc; i++) {
+        // Load the Node of the robot by def argument and add to vector
+        webots::Node* tempNode = supervisor.getFromDef(argv[i]);
+        otherRobotsNodes.push_back(tempNode);
+    }
+
+    // Load config file
+    try {
+        YAML::Node config = YAML::LoadFile("config.yaml");
+        rotations         = config["rotations"].as<std::vector<std::array<double, 4>>>();
+        minDistance       = config["minDistance"].as<double>();
+        zHeight           = config["zHeight"].as<double>();
+    }
+    catch (const YAML::BadFile& e) {
+        std::cerr << e.msg << std::endl;
+        return 1;
+    }
+    catch (const YAML::ParserException& e) {
+        std::cerr << e.msg << std::endl;
+        return 2;
+    }
+
+    webots::Node* tempNode = supervisor.getFromDef("test");
+    xSize                  = tempNode->getField("xSize")->getSFFloat();
+    ySize                  = tempNode->getField("ySize")->getSFFloat();
+
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+
+    // Generate distributions for random number systems in the following loop
+    std::uniform_int_distribution<> xDistrib(0, xSize * 100);
+    std::uniform_int_distribution<> yDistrib(0, ySize * 100);
+    std::uniform_int_distribution<> rotDistrib(0, rotations.size() - 1);
```

Comment:
```suggestion
    std::uniform_real_distribution<> xDistrib(-xSize * 0.5, xSize * 0.5);
    std::uniform_real_distribution<> yDistrib(-ySize * 0.5, ySize * 0.5);
    std::uniform_real_distribution<> rotDistrib(0.0, rotations.size() - 1.0);
```

Reply:
Oh that's much cleaner I didn't realise I could do that